### PR TITLE
Proper PHP 8 support

### PR DESCRIPTION
**Pull request by Girgias**
This PR adds proper PHP 8 support and does a couple of performance optimizations and type fixes.

Also removes irrelevant TSRML macros that are not relevant as of PHP 7.

I will note that there is a massive mess between tab and space indentations. And C files use tabs in general.

### CODE REVIEW

**Comment by Girgias** [comment]
Also, there should probably be distinct jobs for each version of PHP and one which tracks master. I don't know how to do this with GitHub actions but I've got something made for GitLab CI for ext/csv: https://gitlab.com/Girgias/csv-php-extension/-/blob/master/.gitlab-ci.yml

**Comment by illuusio** [comment]
> Also, there should probably be distinct jobs for each version of PHP and one which tracks master. I don't know how to do this with GitHub actions but I've got something made for GitLab CI for ext/csv: https://gitlab.com/Girgias/csv-php-extension/-/blob/master/.gitlab-ci.yml

It's mostly the same. I'm learning Github actions and they are bit clumsy to make as they are in Gitlab also but I'll make PR to track it after we have settle this. Please be free to open Issue for that and this messy tab/space thing. I'm  more spaceman by myself (old school C guy) so and like to use astyle, GNU indent or clang-format not manual formating.

Sorry this is out of PR thing heavily..

**Comment by illuusio** [comment]
Please rebase as there is now new reusable workflow which is better for testing

File: blenc.c [hunk 1]
```diff
@@ -284,9 +284,6 @@ PHP_FUNCTION(blenc_encrypt)
 	        }
 	}
 
-	if(strlen(data) > 0 && data_len == 0) {
```
Comment (illuusio) [inline blenc.c:287]: This one I don't understand? I don't see how it's dead code?

**Comment by illuusio** [comment]
I took advantage to open PR #9 to remove TSRML macros. I kind of a fan of small PR's and they kind of belong here but still I thought that they easier to review as own PR. I cherry-picked your commit and fixed `php_blenc.h` part

**Comment by Girgias** [comment]
> Please rebase as there is now new reusable workflow which is better for testing

Rebased, didn't drop the TSRML commit as I'm getting some conflicts otherwise.

**Comment by illuusio** [comment]
I've made those size_t changes  #12 and added testcase (also added these PHP 7.x) to CI. Could you see @Girgias if I missed any int to size_t conversion.

**Comment by illuusio** [comment]
I'll again splitted dead code removing and key_length stuff to new PR #13.  After that there should be only support for PHP 8.

**Comment by illuusio** [comment]
Closing this to favor PR #4 which imports good stuff here and makes them work form PHP 7.0-8.2. Thanks a lot for the push to make these changes @Girgias!

**Comment by Girgias** [comment]
Happy to have been of help. :-)

**Pull request closed without merging**